src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,98 @@
+using System.Linq;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.EnvironmentInfo;
+using NzbDrone.Common.Messaging;
+using NzbDrone.Core.Download;
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ProviderUpdatedEvent<IDownloadClient>))]
+    [CheckOn(typeof(TrackImportedEvent), CheckOnCondition.FailedOnly)]
```

Comment:
Also, do we really want to trigger this on every track import failed? Wouldn’t the client updates and every 6 hours be good enough?

Reply:
I think so, because the standard client update check will only work if `OutputRootFolders` is populated, but that isn't always the case.  For example with Pneumatic it's never set and with nzbget it's only set if you have a category.

It should only run on `TrackImportFailed` if the health check previously passed.  Similarly it'll only run on the `TrackImported` if the previous health check failed.  The event handling is already there so I don't think this should add any extra overhead.